### PR TITLE
10296 - Modify entity_filter to update AUTH-API entities.state

### PR DESCRIPTION
**Pull request by seeker25**
*Issue #:*[/bcgov/entity/10296](https://github.com/bcgov/entity/issues/10296)

*Description of changes:*
Add in AccountService calls to dissolution and correction.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: queue_services/entity-filer/src/entity_filer/worker.py [hunk 1]
```diff
@@ -180,6 +180,13 @@ async def process_filing(filing_msg: Dict, flask_app: Flask):  # pylint: disable
 
                 elif filing.get('dissolution'):
                     dissolution.process(business, filing, filing_submission, filing_meta)
+                    if business.legal_type in ('SP', 'GP'):
```
Comment (vysakh-menon-aot) [inline queue_services/entity-filer/src/entity_filer/worker.py:183]: We may not need this if condition. HISTORICAL state is applicable for all legal type I think! @lekshmimallika-aot can you confirm?

File: queue_services/entity-filer/src/entity_filer/worker.py [hunk 2]
```diff
@@ -210,6 +217,13 @@ async def process_filing(filing_msg: Dict, flask_app: Flask):  # pylint: disable
 
                 elif filing.get('correction'):
                     filing_submission = correction.process(filing_submission, filing, filing_meta)
+                    if business.legal_type in ('SP', 'GP') and business.state == Business.State.HISTORICAL:
```
Comment (vysakh-menon-aot) [inline queue_services/entity-filer/src/entity_filer/worker.py:220]: Same as above

Comment (seeker25) [reply queue_services/entity-filer/src/entity_filer/worker.py]: Fixed

File: queue_services/entity-filer/src/entity_filer/worker.py (hunk 1 above)
Comment (seeker25) [reply queue_services/entity-filer/src/entity_filer/worker.py]: Fixed

File: queue_services/entity-filer/src/entity_filer/worker.py [hunk 3]
```diff
@@ -210,6 +216,13 @@ async def process_filing(filing_msg: Dict, flask_app: Flask):  # pylint: disable
 
                 elif filing.get('correction'):
                     filing_submission = correction.process(filing_submission, filing, filing_meta)
+                    if business.state == Business.State.HISTORICAL:
```
Comment (lekshmimallika-aot) [inline queue_services/entity-filer/src/entity_filer/worker.py:219]: Which business scenario is this? If the business state is historical, can you file a correction?

Comment (seeker25) [reply queue_services/entity-filer/src/entity_filer/worker.py:219]: The business scenario is when a company is dissolved (by mistake), but staff needs to correct it back to ACTIVE.

Comment (lekshmimallika-aot) [reply queue_services/entity-filer/src/entity_filer/worker.py:219]: I think that action is called restoration. Correction is disabled for a historical entity unless this is a new requirement

Comment (seeker25) [reply queue_services/entity-filer/src/entity_filer/worker.py:227]: I've moved it to restoration.

**Comment by seeker25** [comment]
Closing this ticket, moving to the two others I've created.

**Review by lekshmimallika-aot — APPROVED** [review]

**Pull request closed without merging**